packages/vue/src/pinia.ts

```diff
@@ -0,0 +1,87 @@
+import { addBreadcrumb, getClient, getCurrentScope, getGlobalScope } from '@sentry/core';
+import { addNonEnumerableProperty } from '@sentry/utils';
+
+// Inline PiniaPlugin type
+type PiniaPlugin = (context: {
+  store: {
+    $id: string;
+    $state: unknown;
+    $onAction: (callback: (context: { name: string; after: (callback: () => void) => void }) => void) => void;
+  };
+}) => void;
+
+type SentryPiniaPluginOptions = {
+  attachPiniaState?: boolean;
+  actionTransformer: (action: any) => any;
+  stateTransformer: (state: any) => any;
+};
+
+export const createSentryPiniaPlugin: (options?: SentryPiniaPluginOptions) => PiniaPlugin = (
+  options: SentryPiniaPluginOptions = {
+    attachPiniaState: true,
+    actionTransformer: action => action,
+    stateTransformer: state => state,
+  },
+) => {
+  const plugin: PiniaPlugin = ({ store }) => {
+    options.attachPiniaState &&
+      getGlobalScope().addEventProcessor((event, hint) => {
+        try {
+          // Get current timestamp in hh:mm:ss
+          const timestamp = new Date().toTimeString().split(' ')[0];
+          const filename = `pinia_state_${store.$id}_${timestamp}.json`;
+
+          hint.attachments = [
+            ...(hint.attachments || []),
+            {
+              filename,
+              data: JSON.stringify(store.$state),
+            },
+          ];
+        } catch (_) {
+          // empty
+        }
+
+        return event;
+      });
+
+    store.$onAction(context => {
+      context.after(() => {
+        const transformedAction = options.actionTransformer(context.name);
+
+        if (typeof transformedAction !== 'undefined' && transformedAction !== null) {
+          addBreadcrumb({
```

Comment:
I would add an option to turn off creating breadcrumbs. From experience, some apps have a lot of state updates and breadcrumbs might be very spammy.

Reply:
Updated: https://github.com/getsentry/sentry-javascript/pull/13841/commits/ea7ba2254fe829d180a43b61be6329d54ad00aa1